File: framework/web/src/main/java/top/bettercode/simpleframework/exception/BusinessException.java
package top.bettercode.simpleframework.exception;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ResponseStatus;

/**
 * @author Peter Wu
 */
@ResponseStatus(HttpStatus.OK)
public class BusinessException extends SystemException {

  private static final long serialVersionUID = 1L;

  public BusinessException(String code) {
    super(code);
  }

  public BusinessException(String code, Throwable cause) {
    super(code, cause);
  }

  public BusinessException(String code, Object data) {
    super(code, data);
  }

  public BusinessException(String code, String message) {
    super(code, message);
  }

  public BusinessException(String code, String message, Throwable cause) {
    super(code, message, cause);
  }

  public BusinessException(String code, String message, Object data) {
    super(code, message, data);
  }
}

File: web/src/main/kotlin/top/bettercode/summer/web/exception/BusinessException.kt
package top.bettercode.summer.web.exception

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ResponseStatus

/**
 * @author Peter Wu
 */
@ResponseStatus(HttpStatus.OK)
class BusinessException : SystemException {
    constructor(httpStatusCode: Int) : super(httpStatusCode)
    constructor(httpStatusCode: Int, cause: Throwable?) : super(httpStatusCode, cause)
    constructor(httpStatusCode: Int, data: Any?) : super(httpStatusCode, data)
    constructor(code: String) : super(code)
    constructor(code: String, cause: Throwable?) : super(code, cause)
    constructor(code: String, data: Any?) : super(code, data)
    constructor(code: String, message: String?) : super(code, message)
    constructor(code: String, message: String?, cause: Throwable?) : super(code, message, cause)
    constructor(code: String, message: String?, data: Any?) : super(code, message, data)
    constructor(httpStatusCode: Int, message: String?) : super(httpStatusCode, message)
    constructor(httpStatusCode: Int, message: String?, cause: Throwable?) : super(httpStatusCode, message, cause)
    constructor(httpStatusCode: Int, message: String?, data: Any?) : super(httpStatusCode, message, data)
    constructor(httpStatusCode: Int, code: String, message: String?) : super(httpStatusCode, code, message)
    constructor(httpStatusCode: Int, code: String, message: String?, cause: Throwable?) : super(httpStatusCode, code, message, cause)
    constructor(httpStatusCode: Int, code: String, message: String?, data: Any?) : super(httpStatusCode, code, message, data)
}
